### PR TITLE
Scrum 26 - Design individual players' page

### PR DESCRIPTION
**Pull request by lebenpdx**
Designed individual players' page.

Elements
- Heading with player render, name, and player number
- Small bits of player information
- Links to individual team page and social media pages
- Player image
- Timeline of former teams played on, as well as their badges
- List of honours received during career

chrome_Ks12XegogE
chrome_FqIlp6r5gm

### CODE REVIEW

File: cs464-group-project/src/components/player/PlayerFormerBadges.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        const formerResponse = await getFormerTeams(id);
+
+        setFormerTeams(formerResponse.formerteams);
+      } catch (err) {
+        console.error('Error getting  former teams', err);
+      }
+    };
+
+    fetchData();
+  }, []);
```
Comment (txiong002) [inline cs464-group-project/src/components/player/PlayerFormerBadges.jsx]: gettitng error/warning for missing dependency on line 19

File: cs464-group-project/src/components/player/PlayerFormerTeams.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        const contractResponse = await getPlayerContractsById(id);
+        setContract(contractResponse.contracts);
+        setFormerTeams(formerResponse.formerteams);
+      } catch (err) {
+        console.error('Error getting  former teams or contracts', err);
+      }
+    };
+
+    fetchData();
+  }, []);
```
Comment (txiong002) [inline cs464-group-project/src/components/player/PlayerFormerTeams.jsx]: getting error/warnning for missing dependency on line 25

File: cs464-group-project/src/components/player/PlayerHonours.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      try {
+        const response = await getHonoursById(id);
+        setHonours(response);
+      } catch (err) {
+        console.error('Error getting honours', err);
+      }
+    };
+
+    fetchData();
+  }, []);
```
Comment (txiong002) [inline cs464-group-project/src/components/player/PlayerHonours.jsx]: getting error/warning for missing dependency line 18

**Review by txiong002 — CHANGES_REQUESTED** [review]
Minor: getting some error/warnings on some components for missing dependcy.

Layout Changes Recommendations: move honours into this col
image

You could remove image from the header since you already have pic in cols below:
header_pic

File: cs464-group-project/src/components/player/PlayerFormerBadges.jsx (hunk 1 above)
Comment (lebenpdx) [reply cs464-group-project/src/components/player/PlayerFormerBadges.jsx]: Should be fixed now

File: cs464-group-project/src/components/player/PlayerFormerTeams.jsx (hunk 2 above)
Comment (lebenpdx) [reply cs464-group-project/src/components/player/PlayerFormerTeams.jsx]: Should be fixed now

File: cs464-group-project/src/components/player/PlayerHonours.jsx (hunk 3 above)
Comment (lebenpdx) [reply cs464-group-project/src/components/player/PlayerHonours.jsx]: Should be fixed now

**Comment by lebenpdx** [comment]
I removed the image from the header. But I think I prefer the table at the bottom for now. There are some edge case players who have 10+ honours and it messes with the formatting when placed higher up. I tried to set an overflow but I couldn't get the alignment of it to be as consistent as I wanted.
chrome_SxuaKvC1RY

**Review by txiong002 — CHANGES_REQUESTED** [review]
Mobile View Change: Can you center the name and number when window resize for mobile view: 
image

**Comment by lebenpdx** [comment]
I centered the heading on mobile view, and switched the info area with the player image
chrome_ZBZK40ilek

**Comment by txiong002** [comment]
> I centered the heading on mobile view, and switched the info area with the player image chrome_ZBZK40ilek

the alignment is still off a little:
image

**Comment by lebenpdx** [comment]
This should be better

chrome_jJ6JWLxxCl

**Comment by lebenpdx** [comment]
> > I centered the heading on mobile view, and switched the info area with the player image chrome_ZBZK40ilek
> 
> the alignment is still off a little: image

This should be better

chrome_jJ6JWLxxCl

**Review by txiong002 — CHANGES_REQUESTED** [review]
If no honours to display you should put some text  there, or dont display honours.
image

**Comment by txiong002** [comment]
> > > I centered the heading on mobile view, and switched the info area with the player image chrome_ZBZK40ilek
> > 
> > 
> > the alignment is still off a little: image
> 
> This should be better
> 
> chrome_jJ6JWLxxCl

Alignment looks good now!

**Comment by lebenpdx** [comment]
> If no honours to display you should put some text there, or dont display honours. image

I went with populating the table with "No career honours"
chrome_BSB9RDG5Wg

**Review by txiong002 — APPROVED** [review]
Everything looks good now.  Good work!

**Pull request merged**